File: src/components/info/weapon/non_elemental.ts
import Weapon, { WeaponGroup } from "./_base";
import { EFFECTS as eff } from "../_effect";

let weapon: Weapon[] = [];

weapon.push({
    name: "non-elemental",
    effects: [{ effect: eff.FLOOR_POTENCY, amount: 1.75 }],
});

const NON_ELEMENTAL: WeaponGroup = {
    name: "non-elemental",
    weapons: weapon,
};

export default NON_ELEMENTAL;

File: _data/weapons/non_elemental.py
from typing import List

from ._weapon import Weapon
from ._weapon_groups import WeaponGroup
from effect import EffectTypes as ET, Effect

TYPE = WeaponGroup.NORMAL

weapons: List[Weapon] = []

weapons.append(
    Weapon(
        "non elemental weapon",
        (Effect(ET.FLOOR_POT, 1.75),),
        TYPE,
    )
)
